pilot/pkg/security/authz/model/model.go

```diff
@@ -46,6 +46,8 @@ const (
 	attrConnSNI          = "connection.sni"              // server name indication, e.g. "www.example.com".
 	attrEnvoyFilter      = "experimental.envoy.filters." // an experimental attribute for checking Envoy Metadata directly.
 
+	attrRequestRegexHeader = "request.regex.headers" // header name is surrounded by brackets, e.g. "request.regex.headers[User-Agent]".
```

Comment:
This feels a bit odd to me. Aren't these essentially the "keys"? `regex` here is not a part of the key, its a part of the matcher

Reply:
Are you thinking it would be more appropriate to have a more generic `regex` field, in addition to `values` and `notValues`?

Reply:
> This feels a bit odd to me. Aren't these essentially the "keys"? regex here is not a part of the key, its a part of the matcher

I don't disagree, and ultimately this is similar to the discussion in the linked design doc about supporting regex matching for request paths. In any case, this is what we've been supporting in Maistra for a while now. Ultimately it's just a way to support this without changing current behaviour and with no API change.